### ISSUE TITLE
how to remove "made with chainlit" footer?

### ISSUE THREAD

**Issue by rakshithsajjan** [Chainlit/chainlit#122]
I want to build a product using chain lit but the footer makes it kinda unprofessional. Is there anyway to remove it?

**Comment by willydouhard** [comment]
Not as of now, but we are interested to hear about your use case and requirement to provide the best solution. Can you ping me on discord?

**Comment by gilfernandes** [comment]
If you are on 0.7.0 or later you can hide the footer by using a custom CSS file.

Instructions:

Add this line to section `UI` in `config.toml ` in folder `.chainlit`:

```toml
# Custom CSS file that can be used to customize the UI.
custom_css = '/assets/test.css'
```

In `/assets/test.css` or whichever CSS file you have add this:

```css
a[href*='https://github.com/Chainlit/chainlit'] {
    visibility: hidden;
}
```

**Comment by wxp16** [comment]
hello @gilfernandes , the above version doesn't work for chainlit "0.7.2". The "made with chainlit" footer was still there after setting the custom_css in the config.toml file.  Did I miss something?

**Comment by onepointconsulting** [comment]
@wxp16 Actually I tried out these instructions and it works.

So make sure that you add the property with the css to your .`.config.toml` file:

```
# Specify a CSS file that can be used to customize the user interface.
# The CSS file can be served from the public directory or via an external link.
custom_css = "/public/elastic.css"
```

Then add the file like this:

![image](https://github.com/Chainlit/chainlit/assets/35300398/298a9868-b9e9-4190-976a-d7150de2c0cf)

Then put this in your css file:

a[href*='https://github.com/Chainlit/chainlit'] {
    visibility: hidden;
}

And you get the results:

![image](https://github.com/Chainlit/chainlit/assets/35300398/f1c2df5b-5bcc-441a-ac53-b8a6cb6c55ee)

**Comment by DaviReisVieira** [comment]
Is that actually working? For me, in 0.7.700, no.

**Comment by JaredChung** [comment]
Stopped working for me when I upgraded to latest chainlit==1.0.100

**Comment by enriqueramosg** [comment]
It does still work. However, i added also this rule:
`a img[alt="watermark"] {
    display: none;
}

a[href*='https://github.com/Chainlit/chainlit'] {
    visibility: hidden;
}`

**Comment by snavid** [comment]
it still doesn' t work

**Comment by enriqueramosg** [comment]
Then explain this :)
<img width="1728" alt="Screenshot 2024-01-31 at 11 03 44" src="https://github.com/Chainlit/chainlit/assets/114602227/07915500-e62a-4fd4-99a6-69eb0ee533ab">

**Comment by ggralla** [comment]
In case it's useful, for version `chainlit==1.0.200` this line in the custom CSS file did the trick for me:

```
.watermark {
  display: none !important;
}
```

**Comment by Nikhi-l** [comment]
```
.watermark {
  display: none !important;
}
```
not working, any other workaround?

**Comment by ggralla** [comment]
@Nikhi-l 2 things to check if this still isn't working for you:
1. Confirm that the custom CSS file is being loaded correctly (can use Network tab of Dev Tools of browser)
2. Manually apply styles using "Inspect Element" in Dev Tools of browser to troubleshoot any CSS issues

**Comment by Nikhi-l** [comment]
@ggralla custom css is not loaded.

i have added the custom css file path in config though

**Comment by tonykay** [comment]
I didn't play around too much, but it appeared to me (Chainlit version 1.0.200) that the path has to start `/public`, and I found `.public/` or `/assets/` both failed. That said I didn't investigate too much.

**Comment by Reymond190** [comment]
worked for me with chainlit==1.0.401, i did this:
config.toml:
custom_css = '/public/test.css'

test.css:
.watermark {
  display: none !important;
}

**Comment by kcentric** [comment]
any tip on how to add your own custom logo maybe?

**Comment by Spritan** [comment]
@kcentric https://docs.chainlit.io/customisation/custom-logo-and-favicon

**Comment by tomatyss** [comment]
> any tip on how to add your own custom logo maybe?

Use custom js like 

document.addEventListener('DOMContentLoaded', function() {
    // Function to override the target element content
    function overrideTargetElement() {
      // Select the target element by its class
      const targetElement = document.querySelector('.MuiStack-root.watermark');
      console.log(targetElement);
      
      if (targetElement) {
        // Replacing the content of the target element
        targetElement.innerHTML = `
          <div>
           <p>Some text</p>
          </div>
        `;
  
        // Console verification
        console.log('Custom script executed successfully!');
      } else {
        // Console error if the target element is not found
        console.error('Target element not found!');
      }
    }
  
    // Create a MutationObserver to watch for changes in the DOM
    const observer = new MutationObserver(function(mutationsList, observer) {
      for (const mutation of mutationsList) {
        if (mutation.type === 'childList') {
          // Try to find the target element whenever child nodes are added
          const targetElement = document.querySelector('.MuiStack-root.watermark');
          if (targetElement) {
            // If the target element is found, override its content and stop observing
            overrideTargetElement();
            observer.disconnect();
          }
        }
      }
    });
  
    // Start observing the entire document for child node changes
    observer.observe(document.body, { childList: true, subtree: true });
  
    // Call the function initially in case the element is already present
    overrideTargetElement();
  });

**Comment by AnupamKris** [comment]
@tomatyss will it work even if I toggle the sidebar? or does it revert back to the original?

**Comment by Spritan** [comment]
@AnupamKris this will dowork if you toggle the sidebar

a[href*='https://github.com/Chainlit/chainlit'] {
    color: transparent; /* Hides the original text */
    position: relative;
}

a[href*='https://github.com/Chainlit/chainlit']::after {
    content: "Its me."; /* New text to display */
    color: black; /* Color of the new text */
    position: absolute;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
}

**Comment by Spritan** [comment]
@ThatOneCoder20  havent tried it yet, but best option i can think of is to replace it while keeping he text

**Comment by sasaki-ebase** [comment]
@ThatOneCoder20,

Yes, it's possible to remove the Chainlit logo from the footer while keeping the "Built with" text. Here are the steps:

1. Uncomment `custom_js` in the `config.toml` file.

2. Create a file in the root directory where Chainlit is running, following the same path structure specified in `custom_js`. For example, if `custom_js = "/public/script.js"`. Note that by default `custom_js = "/public/test.js"`.

3. In this JavaScript file, use the following code:

```javascript
document.addEventListener('DOMContentLoaded', function() {
    function overrideFooterElement() {
        const footerElement = document.querySelector("#root > div > div > div > div.MuiStack-root.css-1ylu0bo > div.MuiStack-root.css-zpi9s5 > div > div > div.MuiBox-root.css-mww0i9 > div.MuiStack-root.watermark.css-1705j0v > a");
        if(footerElement) {
            const pElement = footerElement.querySelector('p');
            if(pElement) {
                pElement.textContent = 'example text';
            }
        }
    }

    setTimeout(overrideFooterElement, appropriate_time);
});
```

Replace `appropriate_time` with the amount of time you want to wait before the function is executed.

This code should be implemented after the DOM is generated. I've used `setTimeout` for this, but there are also methods to monitor DOM elements. However, my code does not utilize such methods and may not be the cleanest solution.

One thing to note is that using `DomContentLoaded` alone won't work because this link is dynamically generated. You'll need to use a process like `setTimeout` or a DOM observer.

This is a temporary solution and there may be better ways to handle this. I hope it can serve as a useful reference for you. Let me know if you have any other questions.

**Comment by yuyuyu2118** [comment]
@ThatOneCoder20,

Yes, it's possible to remove the Chainlit logo from the footer while keeping the "Built with" text. Here are the steps:

1. Uncomment `custom_js` in the `config.toml` file.

2. Create a file in the root directory where Chainlit is running, following the same path structure specified in `custom_js`. For example, if `custom_js = "/public/script.js"`. Note that by default `custom_js = "/public/test.js"`.

3. In this JavaScript file, use the following code:

```javascript
document.addEventListener('DOMContentLoaded', function() {
    function overrideFooterElement() {
        const footerElement = document.querySelector("#root > div > div > div > div.MuiStack-root.css-1ylu0bo > div.MuiStack-root.css-zpi9s5 > div > div > div.MuiBox-root.css-mww0i9 > div.MuiStack-root.watermark.css-1705j0v > a");
        if(footerElement) {
            const pElement = footerElement.querySelector('p');
            if(pElement) {
                pElement.textContent = 'example text';
            }
        }
    }

    setTimeout(overrideFooterElement, appropriate_time);
});
```

Replace `appropriate_time` with the amount of time you want to wait before the function is executed.

This code should be implemented after the DOM is generated. I've used `setTimeout` for this, but there are also methods to monitor DOM elements. However, my code does not utilize such methods and may not be the cleanest solution.

One thing to note is that using `DomContentLoaded` alone won't work because this link is dynamically generated. You'll need to use a process like `setTimeout` or a DOM observer.

This is a temporary solution and there may be better ways to handle this. I hope it can serve as a useful reference for you. Let me know if you have any other questions.

**Comment by VioletRaven** [comment]
Any news about how to remove the footer? There is really no easy solution I guess...

**Comment by Leping0uin** [comment]
Hi all ! 
Locate the file index-cf48bedd.js in Lib/site-packages/chainlit/frontend/dist/assets
Locate the watermark and change the value to null.
This should work 
Code after edit : 
`function mae(){const{theme:e}=mn(Ad),t=e==="light"?fSt:pSt;return g.jsx(kt,{mx:"auto",className:"watermark",children:null});}`

**Comment by VioletRaven** [comment]
**Step 1**: Refresh your browser cache or try using Chrome.
**Step 2**: Add the following code to your stylesheet.css after the body tag:
```

.watermark {
    position: relative;
    width: 100%;
    margin: 15px 0;
}

.watermark > a {
    display: none;
}

.watermark > a > p,
.watermark > a > svg {
   display: none;
}

.watermark::after {
    content: " Created by me!" !important;
    visibility: visible !important;
    color: rgb(157, 157, 155) !important;
    font-family: Inter !important;
    font-size: 14px !important;
    position: absolute !important;
    text-align: center !important;
    left: 50%;
    width: 100%;
    height: auto;
    transform: translateX(-50%);
}

.watermark::before {
    content: " ";
    position: absolute;
    left: 50%;
    transform: translateX(-130px);
    background-image: url('./your_logo.png'); /* Relative path to your custom emoji */
    background-size: contain; /* Adjust as needed */
    background-repeat: no-repeat;
    padding-left: 1.5em; /* Add padding to make space for the logo */
    height: 1em; /* Adjust size as needed */
    vertical-align: middle; /* Align with text */

} . 
```
This setup works with my logo, but you might need to adjust it for your specific logo.

Thanks to @EmanuelaPau on github for the solution!

**Comment by Sriram-93** [comment]
i tried all these methods but its not working,anyone give some tips to implement it

**Comment by hexart** [comment]
try these code, it works on my project.
`.watermark a p, .watermark a svg{
    display: none;
}`

**Comment by desaxce** [comment]
Solved with https://github.com/Chainlit/chainlit/issues/122#issuecomment-1949191665

**Comment by Sriram-93** [comment]
give file structure for this

**Comment by Heiden133** [comment]
Hi any updates on this? In my case, it seems the custom css or js is not being properly recognised by Chainlit at all - I don't see it being loaded in network tab. I've tried also to provide CSS as URL - still nothing.

**Comment by qasim29** [comment]
To efficiently detect and remove the watermark instead of using setTimeout repeatedly

>     function observeAndHideWatermark() {
>       const widget = document.querySelector("#chainlit-copilot");
> 
>       if (widget && widget.shadowRoot) {
>         const observer = new MutationObserver(() => {
>           const watermark = widget.shadowRoot.querySelector("a.watermark");
>           if (watermark) {
>             watermark.style.display = "none";
>           }
>         });
> 
>         observer.observe(widget.shadowRoot, { childList: true, subtree: true });
>       }
>     }
> 
>     observeAndHideWatermark();

**Comment by whiteboardmonk** [comment]
Someone trying this on the latest version the following code works for me

`function observeAndHideWatermark() {
  // Hide existing watermark immediately
  const existingWatermark = document.querySelector("a.watermark");
  if (existingWatermark) {
    existingWatermark.style.display = "none";
  }

  // Set up observer for future watermarks
  const observer = new MutationObserver(() => {
    const watermark = document.querySelector("a.watermark");
    if (watermark && watermark.style.display !== "none") {
      watermark.style.display = "none";
    }
  });

  observer.observe(document.body, { 
    childList: true, 
    subtree: true,
    attributes: true,
    attributeFilter: ['style']
  });
}

// Run immediately and also after DOM is fully loaded
observeAndHideWatermark();
if (document.readyState === 'loading') {
  document.addEventListener('DOMContentLoaded', observeAndHideWatermark);
}`

**Comment by 123sameer456** [comment]
This code is working
- my Chainlit, version 2.5.5
- make sure you remove hash (#) sign from the line in [UI] portion for paths variable first.
- here is my folder location: custom_js = "/public/test.js"

document.addEventListener('DOMContentLoaded', function () {
    // Function to override the target element content
    function overrideTargetElement() {
        // Select the target element by its class
        const targetElement = document.querySelector('.MuiStack-root.watermark') || document.querySelector('.watermark');
        console.log(targetElement);

        if (targetElement) {
            // Hide the watermark by setting display to none
            targetElement.style.display = 'none';

            // Console verification
            console.log('Watermark hidden successfully!');
        } else {
            // Console error if the target element is not found
            console.error('Target element not found!');
        }
    }

    // Create a MutationObserver to watch for changes in the DOM
    const observer = new MutationObserver(function (mutationsList, observer) {
        for (const mutation of mutationsList) {
            if (mutation.type === 'childList') {
                // Try to find the target element whenever child nodes are added
                const targetElement = document.querySelector('.MuiStack-root.watermark') || document.querySelector('.watermark');
                if (targetElement) {
                    // If the target element is found, hide it and stop observing
                    overrideTargetElement();
                    observer.disconnect();
                }
            }
        }
    });

    // Start observing the entire document for child node changes
    observer.observe(document.body, { childList: true, subtree: true });

    // Call the function initially in case the element is already present
    overrideTargetElement();
});

**Comment by Sukanya41455** [comment]
@123sameer456  This didn't work for me.
